source/Octopus.Server.Client/Model/Endpoints/KubernetesAgentUpgradeInformationResource.cs

```diff
@@ -0,0 +1,8 @@
+namespace Octopus.Client.Model.Endpoints;
+
+public record KubernetesAgentUpgradeInformationResource
```

Comment:
I don't love this name, can you think of any better ones?

Reply:
What's this meant to store?

Reply:
Basically just those two properties: `UpgradeStatus` which can be `NoUpgrades`, `UpgradeAvailable`, `UpgradeSuggested`, and `UpgradeRequired`. These values are based on how out-of-date the current version is to the suggested version. Eg: if a version is a major version or more behind, it's `UpgradeRequired`, if it's a minor version out it's `UpgradeSuggested` etc.
The other property is `UpgradeLocked`. If this is set by the user in the target connectivity page, the target won't be updated until it's unlocked.

Reply:
I've moved the UpgradeLocked and UpgradeStatus properties as we discussed. I've applied the same changes to the Server PR and I think it works pretty well. 👍

Reply:
It looks like we [got two thumbs up](https://octopusdeploy.slack.com/archives/C04LWDDAC6L/p1713340955931229) for this approach from #team-backend-foundations.